server/annotation/middleware.ts

```diff
@@ -0,0 +1,102 @@
+import type { Request, Response, NextFunction } from "express";
+import { isValidObjectId } from "mongoose";
+import AnnotationCollection from "./collection";
+import AnnotationModel from "./model";
+
+/**
+ * Checks if the lecture ID is a valid lecture
+ */
+const isValidLectureId = async (
```

Comment:
For this and the same function in post, we should probably just have this code once in Lecture middleware once that is implemented.